gym/envs/doom/doom_env.py

```diff
@@ -66,29 +118,46 @@ def _step(self, action):
             return np.zeros(shape=self.observation_space.shape, dtype=np.uint8), 0, True, {}
 
     def _reset(self):
-        self._closed = False
-        package_directory = os.path.dirname(os.path.abspath(__file__))
+        # Reset is called on first step() after level is finished
+        # or when change_level() is called. Returning if neither have been called to
+        # avoid resetting the level twice
+        if self.find_new_level:
+            return
+
+        # .init() called, .close() not called, and on same level
+        if self.is_initialized and not self._closed and self.previous_level == self.level:
+            self._start_episode()
+            return self.game.get_state().image_buffer.copy()
+
+        # Closing if is_initialized
+        if self.is_initialized:
+            self.is_initialized = False
+            self.game.close()
+            self.game = DoomGame()
+
+        # Loading Paths
+        if not self.is_initialized:
+            self.game.set_vizdoom_path(self.loader.get_vizdoom_path())
+            self.game.set_doom_game_path(self.loader.get_freedoom_path())
 
         # Common settings
-        self.loader = Loader()
-        self.game = DoomGame()
-        if self.config != '':
-            self.game.load_config(os.path.join(package_directory, 'assets/%s' % self.config))
-        self.game.set_vizdoom_path(self.loader.get_vizdoom_path())
-        self.game.set_doom_game_path(self.loader.get_freedoom_path())
-        if self.scenario != '':
-            self.game.set_doom_scenario_path(self.loader.get_scenario_path(self.scenario))
-        if self.map != '':
-            self.game.set_doom_map(self.map)
-        self.game.set_doom_skill(self.difficulty)
+        self._closed = False
+        self.game.load_config(os.path.join(self.doom_dir, 'assets/%s' % DOOM_SETTINGS[self.level][CONFIG]))
```

Comment:
It does only call _start_episode, but the game must be initiated (so game.init() has been called), and not closed (game.close() not called), and the same level is played.

```python
        if self.is_initialized and not self._closed and self.previous_level == self.level:
            self._start_episode()
            return self.game.get_state().image_buffer.copy()
```

Otherwise, VizDoom needs to launch another FreeDoom process and those calls are required to build the parameters. I can add the set_() calls to __init__ for the base class, and subclass reset() for the meta class, but I won't really change the logic that I need to set those parameters if the game is not init, or if the level needs to be changed.